### PR TITLE
Datums Readme (SBO47-54)

### LINKED ISSUE #21218: Weeaboo Katana
(WEB REPORT BY: jalis REMOTE: 206.221.180.138:7777)
# Revision
814ef849a66e22865acbe4f63811d3c43479447f

# Description
Description of the weeaboo katana is not intuitive. Description should be changed to say "ALT CLICK on the Sword" and not Alt Click to teleport attack. Maybe change the function for teleporting to the Z key bind for ease of use?
# Steps to Reproduce
Try to alt click on a mob
# What you Expected
To do a teleport attack
# What Actually Happened
Nothing because item description is worded poorly.

### LINKED ISSUE #21288: Crazed weeaboos get double objectives
When a crazed weaboo is selected, they sometimes spawn with double objectives, so they'll have more objectives then they ought too. Not sure what circumstances cause this, but its happened a few times now.

### LINKED ISSUE #21612: Late join traders drafted as weeb use trader spawn
#### Description of issue
- late join as trader with weeb enabled
- get drafted as weeb
- spawn as weeb flying through space

### LINKED ISSUE #21624: On Rare Occasions, Wizards May Spawn In Shuttle Without Equipment
I've seen this on 3 maps, pretty sure it's related to crashing after applying for wizard, so you don't have a mind when it tries to put you in and probably runtimes

### PR DESCRIPTION
**Pull request by Kurfursten**
Adds a readme file for a basic overview of how datums work, and also

* Converted latejoin threat to refund_threat (from cap at 100) and added logging
* Latejoin weeaboo no longer forges objectives again (fixes #21288)
* Improves weeb sword examine text (fixes #21218)
* Traders cannot be weeb role (fixes #21612)
* Refactored RoleMobDestroyed to actually work
* Vampires no longer menace or cloak while dead (reported to me via ahelps)
* A mode can be refunded if no applications were valid, this is also logged
* Fixed a bug where a non-ghosted but dead applicant would get passed over for from-ghost antags
* Fixed a bug where wizards after the first would not be dressed, moved to the den, or named. fixes #21624
* Fixed a bug where a forced latejoin/midround ragin mages would not calculate threat cost properly.
* Threat log more verbose about what kind of rule spent threat

**Important note for transparency** the first change on the list has implications for overall threat generation, mostly in the case that officers latejoin immediately at the start of a round where little or no threat was spent right away. In the old functionality, this would allow threat to go above threat level (e.g.: 23 threat, none spent extended, 2 officers join raising it to 43 threat / 23 threat level). The new functionality uses refund_threat meaning it will not exceed the threat level via latejoins. We can change this and use create_threat() instead if you think it should raise the threat_level!

🆑 
* bugfix: Fixed a detection issue that prevented regaining threat when an antagonist was destroyed. Note that they need to be entirely destroyed, cremate that head!
* tweak: The weeaboo sword's examine text is now more clear about what alt clicking does (it toggles your teleport attack on/off)
* bugfix: Latejoin weeaboos should no longer get double objectives
* bugfix: When a latejoin role that refunds threat joins, it is now properly logged.
* tweak: Latejoin roles will not refund threat over the threat_level
* rscdel: Traders can no longer quality for the Crazed Weeaboo role, to prevent latejoining in space.
* bugfix: Vampires no longer cloak or menace while dead.
* rscadd: A mode can refund its threat now if none of the applicants were valid.
* bugfix: Fixed a bug where a dead but not ghosted player would be ignored for from_ghosts antag roles even after submitting an application.
* bugfix: Fixed a bug where later ragin' mages would not spawn in the wizard den, get a book, robes, name, etc.
* bugfix: Fixed a bug where forced ragin' mages would not calculate threat properly.

### CODE REVIEW

**Comment by SonixApache** [comment]
Not sure how you fixed the latejoin double objective, but good job.

**Comment by Kurfursten** [comment]
latespawn() already calls role's forge objectives, so it was unnecessary to have it a second time called by the latejoin ruleset (this might be worth including in my readme...)

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -110,9 +110,9 @@
 		var/cooldowncalculated = round((teleportcooldown - world.time)/10)
 		message += "Your steel has unleashed it's dark and unwholesome power, so it's tapped out right now. It'll be ready again in [cooldowncalculated] seconds."
 	if(active)
-		message += " Alt-click it to disable your teleport power!</span>"
+		message += " Alt-click it to stalk your prey without teleporting like a Hunter x Hunter!</span>"
 	else
-		message += " Alt-click it to teleport behind those who wish to Kill la Kill you!</span>"
+		message += " Alt-click it enable teleporting toward your Kill la Kill!</span>"
```
Comment (MadmanMartian) [inline code/game/objects/items/weapons/weaponry.dm]: Alt-click it to enable teleporting towards your [whatever that weeb reference is]

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -137,6 +137,8 @@
 			federation = ticker.mode.CreateFaction(/datum/faction/wizard, null, 1)
 		federation.HandleRecruitedRole(newWizard)//this will give the wizard their icon
 		newWizard.Greet(GREET_ROUNDSTART)
+		spawn(1)
```
Comment (MadmanMartian) [inline code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: There HAS TO BE A **BETTER _WAY_**

File: code/game/objects/items/weapons/weaponry.dm [hunk 3]
```diff
@@ -110,9 +110,9 @@
 		var/cooldowncalculated = round((teleportcooldown - world.time)/10)
 		message += "Your steel has unleashed it's dark and unwholesome power, so it's tapped out right now. It'll be ready again in [cooldowncalculated] seconds."
 	if(active)
-		message += " Alt-click it to disable your teleport power!</span>"
+		message += " Alt-click it to stalk your prey without teleporting like a Hunter x Hunter!</span>"
```
Comment (jknpj) [inline code/game/objects/items/weapons/weaponry.dm]: MUH REFERENCES

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (jknpj) [inline code/game/objects/items/weapons/weaponry.dm]: MUH

File: code/game/objects/items/weapons/weaponry.dm (hunk 3 above)
Comment (Kurfursten) [reply code/game/objects/items/weapons/weaponry.dm]: Pretty sure this is the only place in the game where it's unarguably acceptable to make references to anime

**Comment by Kurfursten** [comment]
Note to self: autotator claims in comments that it waits 15 minutes between tots, but due to process ticks it waits 30 minutes

File: code/datums/gamemode/role/role.dm [hunk 4]
```diff
@@ -229,7 +230,17 @@
 	return
 
 /datum/role/proc/process()
-	return
+	var/mob/M = antag.current
+	if(!destroyed)
+		if(!M)
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/role.dm:235]: Shouldn't it be !M || M.gcDestroyed? I assume you're looking for if the mob no longer exists

Comment (Kurfursten) [reply code/datums/gamemode/role/role.dm:235]: We actually have a new role-level destroyed var now. (see L97)

Comment (MadmanMartian) [reply code/datums/gamemode/role/role.dm:235]: That doesn't answer my question. Are you looking to see if the owner of this role was destroyed or not?

Comment (Kurfursten) [reply code/datums/gamemode/role/role.dm:235]: That's right, although the owner mob can change (e.g.: person gets debrained, they are now inside the brain mob)

Although it is near inconceivable that somehow the mob could be destroyed and later the role owner might have a mob again, I wanted to have a case for that

**Pull request merged**